### PR TITLE
Fix json server error return type to always be `int`

### PR DESCRIPTION
**Pull request by alexmerlin**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | yes

### Description
Fixed an error thrown during Psalm analysis.
> Error: src/Error.php:67:13: DocblockTypeContradiction: Cannot resolve types for $code - docblock-defined type int does not contain float (see https://psalm.dev/155

Also:
* added missing return type declarations for the function's in `src/Error.php`
* added missing `/.phpcs-cache` to `.gitignore` so it does not get committed to the repo

### CODE REVIEW

File: src/Error.php [hunk 1]
```diff
@@ -59,10 +59,10 @@ public function __construct($message = null, $code = self::ERROR_OTHER, $data =
      *
      * If the error code is 0, it will be set to -32000 (ERROR_OTHER).
      *
-     * @param  int $code
+     * @param mixed $code
      * @return self
      */
-    public function setCode($code)
+    public function setCode($code): Error
```
Comment (samsonasik) [inline src/Error.php]: add return type is bc break as class is not final, see https://3v4l.org/d8L2h

Comment (alexmerlin) [reply src/Error.php]: fixed

File: src/Error.php [hunk 2]
```diff
@@ -146,7 +148,7 @@ public function getData()
      *
      * @return array
      */
-    public function toArray()
+    public function toArray(): array
```
Comment (samsonasik) [inline src/Error.php]: this also bc break

Comment (alexmerlin) [reply src/Error.php]: fixed

**Review by Ocramius — APPROVED** [review]
Thanks @alexmerlin!

**Pull request merged**